metrics/web/test/auth/presentation/state/auth_notifier_test.dart

```diff
@@ -235,56 +248,110 @@ void main() {
 
-      authNotifier.signInWithEmailAndPassword(email, password);
+      await authNotifier.signInWithEmailAndPassword(email, password);
 
       expect(authNotifier.authErrorMessage, isNotNull);
     });
 
-    test(".authErrorMessage is populated when GoogleSignInUseCase throws", () {
-      when(googleSignInUseCase.call()).thenThrow(authException);
+    test(
+      ".emailErrorMessage is populated when the email-related error occurred during the sign-in process",
+      () async {
+        when(signInUseCase.call(any)).thenThrow(emailAuthException);
 
-      authNotifier.signInWithGoogle();
+        await authNotifier.signInWithEmailAndPassword(email, password);
 
-      expect(authNotifier.authErrorMessage, isNotNull);
-    });
+        expect(authNotifier.emailErrorMessage, isNotNull);
+      },
+    );
 
     test(
-      ".signInWithEmailAndPassword() clears the authentication error message on a successful sign in",
-      () {
-        const invalidEmail = 'email@mail.mail';
-        const invalidPassword = 'password';
-        final invalidCredentials = UserCredentialsParam(
-          email: Email(invalidEmail),
-          password: Password(invalidPassword),
-        );
+      ".passwordErrorMessage is populated when the password-related error occurred during the sign-in process",
+      () async {
+        when(signInUseCase.call(any)).thenThrow(passwordAuthException);
+
+        await authNotifier.signInWithEmailAndPassword(email, password);
+
+        expect(authNotifier.passwordErrorMessage, isNotNull);
+      },
+    );
+
+    test(
+      ".authErrorMessage is populated when GoogleSignInUseCase throws",
+      () async {
+        when(googleSignInUseCase.call()).thenThrow(authException);
+
+        await authNotifier.signInWithGoogle();
+
+        expect(authNotifier.authErrorMessage, isNotNull);
+      },
+    );
 
+    test(
+      ".signInWithEmailAndPassword() clears the authentication error message on a successful sign in",
+      () async {
         when(signInUseCase.call(invalidCredentials)).thenThrow(authException);
 
-        authNotifier.signInWithEmailAndPassword(invalidEmail, invalidPassword);
+        await authNotifier.signInWithEmailAndPassword(
+          invalidEmail,
+          invalidPassword,
+        );
 
         expect(authNotifier.authErrorMessage, isNotNull);
 
-        authNotifier.signInWithEmailAndPassword(
-            'valid_email@mail.mail', 'valid_password');
+        await authNotifier.signInWithEmailAndPassword(email, password);
 
         expect(authNotifier.authErrorMessage, isNull);
       },
     );
 
+    test(
+      ".signInWithEmailAndPassword() clears the authentication email error message on a successful sign in",
+      () async {
+        when(signInUseCase.call(invalidCredentials))
+            .thenThrow(emailAuthException);
+
+        await authNotifier.signInWithEmailAndPassword(
+            invalidEmail, invalidPassword);
```

Comment:
Would it be more pretty and readable if we'll add a trailing comma in such situations?
```suggestion
        await authNotifier.signInWithEmailAndPassword(
          invalidEmail,
          invalidPassword,
        );
```